packages/liveblocks-react/src/factory.tsx

```diff
@@ -280,45 +319,44 @@ export function createRoomContext<
     // Note: We'll hold on to the initial value given here, and ignore any
     // changes to this argument in subsequent renders
     const frozen = useInitial({
+      instanceId,
       initialPresence,
       initialStorage,
       unstable_batchedUpdates,
-      shouldInitiallyConnect:
-        shouldInitiallyConnect === undefined
-          ? typeof window !== "undefined"
-          : shouldInitiallyConnect,
+      autoConnect,
     });
 
-    const [room, setRoom] = React.useState<
-      Room<TPresence, TStorage, TUserMeta, TRoomEvent>
-    >(() =>
-      client.enter(roomId, {
+    const [{ room }, setRoomLeavePair] = React.useState(() =>
+      stableEnterRoom(frozen.instanceId, roomId, {
         initialPresence: frozen.initialPresence,
         initialStorage: frozen.initialStorage,
-        shouldInitiallyConnect: frozen.shouldInitiallyConnect,
+        autoConnect: false,
```

Comment:
is this intentional? Should it be frozen.autoConnect?

Reply:
Yep, see below!